src/main/java/com/algorand/algosdk/v2/client/model/IndexerStateProofMessage.java

```diff
@@ -0,0 +1,68 @@
+package com.algorand.algosdk.v2.client.model;
+
+import java.util.Objects;
+
+import com.algorand.algosdk.util.Encoder;
+import com.algorand.algosdk.v2.client.common.PathResponse;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class IndexerStateProofMessage extends PathResponse {
+
+    /**
+     * (b)
+     */
+    @JsonProperty("block-headers-commitment")
+    public void blockHeadersCommitment(String base64Encoded) {
+        this.blockHeadersCommitment = Encoder.decodeFromBase64(base64Encoded);
+    }
+    public String blockHeadersCommitment() {
+        return Encoder.encodeToBase64(this.blockHeadersCommitment);
+    }
+    public byte[] blockHeadersCommitment;
+
+    /**
+     * (f)
+     */
+    @JsonProperty("first-attested-round")
+    public java.math.BigInteger firstAttestedRound;
```

Comment:
are these code generated? I think long type should be sufficient.

Reply:
This one is generated code. I think it means the type was tagged with `uint64`, so the generator uses a BigInteger.

Reply:
I see.